router/connection.go

```diff
@@ -11,6 +11,10 @@ import (
 	"time"
 )
 
+type ConnectionSender interface {
+	SendTCP(msg []byte)
+}
+
```

Comment:
I wouldn't bother with an interface with that; just cast to LocalConnection in the one call site.